### PR TITLE
fix: bounty tests, superceeds #799

### PR DESCRIPTION
**Pull request by shrugs**
replaces #799

### CODE REVIEW

File: test/Bounty.test.js [hunk 1]
```diff
@@ -1,113 +1,89 @@
...
-
-        try {
-          await bounty.claim(targetAddress, { from: researcher });
-          assert.isTrue(false); // should never reach here
-        } catch (error) {
-          let reClaimedBounty = await bounty.claimed.call();
-          assert.isFalse(reClaimedBounty);
-        }
-        try {
-          await bounty.withdrawPayments({ from: researcher });
```
Comment (shrugs) [inline test/Bounty.test.js]: no idea what this was testing; the transaction actually goes through but obviously withdraws 0 wei. this is covered in pullpayment test, so I removed it here

File: test/Bounty.test.js [hunk 2]
```diff
@@ -1,113 +1,89 @@
...
+      const balance = await ethGetBalance(this.bounty.address);
+      balance.should.be.bignumber.eq(reward);
+    });
 
-  it('empties itself when destroyed', async function () {
-    let owner = accounts[0];
-    let reward = web3.toWei(1, 'ether');
-    let bounty = await SecureTargetBounty.new();
-    await sendReward(owner, bounty.address, reward);
+    it('empties itself when destroyed', async function () {
```
Comment (nventuro) [inline test/Bounty.test.js]: Not sure if we should be testing this: this is not `Bounty`'s code, but `Destructible`'s.

File: test/Bounty.test.js [hunk 3]
```diff
@@ -1,113 +1,89 @@
...
+      this.bounty = await SecureTargetBounty.new({ from: owner });
+    });
 
-contract('Bounty', function (accounts) {
-  it('sets reward', async function () {
-    let owner = accounts[0];
-    let reward = web3.toWei(1, 'ether');
-    let bounty = await SecureTargetBounty.new();
-    await sendReward(owner, bounty.address, reward);
+    it('sets reward', async function () {
```
Comment (nventuro) [inline test/Bounty.test.js]: I'd remove this `it` altogether, it's already being tested for in all other test cases (and if we were to add it, we should have it on both `describe`s).

File: test/Bounty.test.js [hunk 4]
```diff
@@ -1,113 +1,89 @@
...
-          assert.equal(reward, updatedBalance.toNumber());
-        }
-      };
-      await bounty.createTarget({ from: researcher });
-      await awaitEvent(event, watcher);
+      const result = await this.bounty.createTarget({ from: researcher });
+      const event = await expectEvent.inLogs(
+        result.logs,
+        'TargetCreated'
+      );
```
Comment (nventuro) [inline test/Bounty.test.js]: nit, but why the newlines? The same line on the other `it` doesn't have them.

File: test/Bounty.test.js [hunk 5]
```diff
@@ -1,113 +1,89 @@
...
+      const event = await expectEvent.inLogs(result.logs, 'TargetCreated');
+
+      const targetAddress = event.args.createdAddress;
+      await sendReward(owner, this.bounty.address, reward);
+
+      const balance = await ethGetBalance(this.bounty.address);
+      balance.should.be.bignumber.eq(reward);
+
+      await this.bounty.claim(targetAddress, { from: researcher });
+      const claim = await this.bounty.claimed.call();
```
Comment (nventuro) [inline test/Bounty.test.js]: Can't you write `.claimed()`?

File: test/Bounty.test.js [hunk 6]
```diff
@@ -1,113 +1,89 @@
...
+
+      claim.should.eq(true);
+
+      await this.bounty.withdrawPayments({ from: researcher });
+      const updatedBalance = await ethGetBalance(this.bounty.address);
+      updatedBalance.should.be.bignumber.eq(0);
```
Comment (nventuro) [inline test/Bounty.test.js]: We should also test:

a) that the researcher got the reward (curr - prev balances = reward)
b) that the bounty is no longer payable after claiming

File: test/Bounty.test.js (hunk 3 above)
Comment (shrugs) [reply test/Bounty.test.js]: I think it's ok to be explicit here. I've refactored the tests in the next commit to make this cleaner

**Pull request merged**